worksheets.rb

```diff
@@ -0,0 +1,170 @@
+## Organize by Driver_ID
+## Nest date
+## Then: cost, rating, rider_id (not in any particular order)
+
+# DATA STRUCTURE
+# drivers hash {
+#     each_driver [
+#       { info_hash }
+#     ]
+# }
+
+drivers = {
+  DR0001: [
+    {
+      date: "3rd Feb 2016",
+      rider_id: "RD0003",
+      cost: 10,
+      rating: 3,
+    },
+    {
+      date: "3rd Feb 2016",
+      rider_id: "RD0015",
+      cost: 30,
+      rating: 4,
+    },
+    {
+      date: "5th Feb 2016",
+      rider_id: "RD0003",
+      cost: 45,
+      rating: 2,
+    },
+  ],
+  DR0002: [
+    {
+      date: "3th Feb 2016",
+      rider_id: "RD0073",
+      cost: 25,
+      rating: 5,
+    },
+    {
+      date: "4th Feb 2016",
+      rider_id: "RD0013",
+      cost: 15,
+      rating: 1,
+    },
+    {
+      date: "5th Feb 2016",
+      rider_id: "RD0066",
+      cost: 35,
+      rating: 3,
+    },
+  ],
+  DR0003: [
+    {
+      date: "4th Feb 2016",
+      rider_id: "RD0066",
+      cost: 5,
+      rating: 5,
+    },
+    {
+      date: "5th Feb 2016",
+      rider_id: "RD0003",
+      cost: 50,
+      rating: 2,
+    },
+  ],
+  DR0004: [
+    {
+      date: "3rd Feb 2016",
+      rider_id: "RD0022",
+      cost: 5,
+      rating: 5,
+    },
+    {
+      date: "4th Feb 2016",
+      rider_id: "RD0022",
+      cost: 10,
+      rating: 4,
+    },
+    {
+      date: "5th Feb 2016",
+      rider_id: "RD0073",
+      cost: 20,
+      rating: 5,
+    },
+  ],
+}
+
+highest_earner = {
+  driver: nil,
+  earnings: 0,
+}
+
+highest_rated = {
+  driver: nil,
+  ratings: 0,
+}
+
+# method for each driver's earnings
+def driver_earnings(input_data)
+  driver_earnings = input_data.map do |ride|
+    ride[:cost]
+  end
+
+  return driver_earnings
+end
+
+# method for each driver's ratings
+def driver_ratings(input_data)
+  driver_ratings = input_data.map do |ride|
+    ride[:rating]
+  end
+
+  return driver_ratings
+end
+
+# Welcome message
+puts "Welcome to the Rideshare Data Program!"
+puts "by Hana C\n"
+
+# loop to access nested information
+drivers.each do |driver, rides|
+  puts "\nDriver: #{driver}"
+
+  # number of rides for each driver
+  puts " Total amount of rides: #{rides.length}"
+
+  # total amount of money earned for each driver
+  earnings = driver_earnings(rides).sum
+  puts " Total amount of earnings: \$#{earnings}"
+
+  ## conditional for highest earner
+  if earnings > highest_earner[:earnings]
+    highest_earner[:driver] = driver
+    highest_earner[:earnings] = earnings
+  end
+
+  # average rating for each driver
+  ratings = driver_ratings(rides).sum / driver_ratings(rides).length
+  puts " Average rating: #{ratings}"
```

Comment:
I notice that you used integer division for the average rating, and lost a little precision.